crates/oxc_semantic/src/builder.rs

```diff
@@ -1765,6 +1765,18 @@ impl<'a> SemanticBuilder<'a> {
                 self.function_stack.push(self.current_node_id);
                 self.make_all_namespaces_valuelike();
             }
+            AstKind::FunctionBody(_) => {
```

Comment:
Can you add test cases for `let i = 1; const foo = () => () => { i++ }`?

Reply:
The current implementation broke this test 🥲. I add and fix it in https://github.com/oxc-project/oxc/pull/5161/commits/66a1b9d4680f8c35a9d4d4e8d05d661a8dd6c74f